packages/peregrine/src/hooks/useWindowSize.js

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect } from 'react';
+
+const getSize = () => {
+    return {
+        innerHeight: window.innerHeight,
+        innerWidth: window.innerWidth,
+        outerHeight: window.outerHeight,
+        outerWidth: window.outerWidth
+    };
+};
+
+/**
+ * A hook that will return inner and outer height and width values whenever
+ * the window is resized.
+ */
+export const useWindowSize = () => {
```

Comment:
After talking with @jimbo we're going to add this to a root context so we don't get the awful side effect of adding a new event listener whenever a component tries to use this.

So basically components will subscribe to the context rather than trying to use this hook directly.

Reply:
🎉 

Right on; most components should call a hook that obtains window size from context, but we'll need to provide that context near the root of the app, and so all the code you've written here will be still be used to attach the listener and set the size in state.